README
Flexibility - version 1.0-rl

-----------------------------------------------------------------------------------------------------------
ABOUT
-----------------------------------------------------------------------------------------------------------

"Flexibility" is a HTML5/CSS3/jQuery based frontend MODx Revolution template based on the "MODX Boilerplate".
With this package you will have a fully functional website with a contact form, slider and a image gallery.

"Flexibility" is designed and coded by Menno Pietersen (blog: http://designfromwithin.com - Twitter: ThaClown)

NOTE:
"MODX Boilerplate" is created by Anselm Hannemann, and it is not a direct clone of the original HTML5Boilerplate by Paul Irish.


-----------------------------------------------------------------------------------------------------------
QUICKSTART        PLEASE READ THIS 
-----------------------------------------------------------------------------------------------------------

1. Install MODx Revolution on your website.

2. Backup the "ht.access" file in from root folder to your desktop and remove the "ht.access" file from your website root (your root is probably something like: "your_url/public_html/").

3. Open the "ht.access" file from your desktop (with notepad or any code editor) and save it as ".htaccess".

4. Copy the new "ht.access" file to your website root.

5. Enter your MODX Revolution manager ("your_url/manager"): Go to "System" > "System Settings" > Click on "Filter by area" and select "Friendly URL" > For "Use Friendly URL's" and "Use Friendly Alias Path" choose "Yes".

6. Download the latest Flexibility version from github (https://github.com/DESIGNfromWITHIN/Flexibility) upload the .transport.zip (do not unzip this file) file to:
<your_website_root>/core/packages/

7. Install the "Flexibility" package: Go to "System" > "Package Management" > "Add New Package" > "Search Locally for Packages" > "Yes".

THAT IS ALL!
(to be sure clear your cache > "Site" > "Clear Cache")


-----------------------------------------------------------------------------------------------------------
ADDING AND CHANGING CONTENT
-----------------------------------------------------------------------------------------------------------

Your MODX website needs three main thing:

1. A Logo.
- Add this under "Resources" > "Site settings" > "Template Variables" > "Logo"

2. An e-mail adress where you want to get mail from the contact form.
- Add this under "Resources" > "Site settings" > "Template Variables" > "E-mail adress"

3. Your adress (remove the default text to keep it empty).
- Add this under "Resources" > "Site settings" > "Template Variables" > "Adress"


And three optional things (only if you want a slider and/or gallery and want custom template colors):

1. Slide content for your website slider.
- Add this under "Resources" > "Site settings" > "Template Variables" > "Slider items"

2. Images and (optional) descriptions for your gallery.
- Add this under "Components" > "Gallery" > "Create Album"


I general you can find all website settings under:

"Resources" > 'select the page you want'
"Resources" > 'select the page you want' > "Template Variables"
"System" > "System Settings" > "Filter by area" > "Site"


-----------------------------------------------------------------------------------------------------------
CONTENT:
-----------------------------------------------------------------------------------------------------------

- Packages: formit, gallery, getresources, if, tinymce, wayfinder
- Templates: Basic page template, Contact page template, Gallery page template, Homepage template, Page settings template
- Template Variables: tv.email_adress, tv.logo, tv_sidebar, tv_slider, tv_tagline, tv_tagline_link, tv_taline_link_text, tv_tagline_option
- Cunks: chunk_contact_form, chunk_email_adress, email_adress_tpl, GalAlbumRowTpl, GalItemThumb, GalleryAlbumTpl, chunk_email_form, logo_tpl, settings_page_id, thank_you_page_id
- Directories: [base_path}/assets
- HTML5 Boilerplate HTML Body
- jQuery 1.6
- basic CSS with reset
- .htaccess File (MODX + Boilerplate)
- favicon + apple touch icons
- Modernizr with HTML5Shim + Modernizr.load/yepnope.js (JS-Loadhelper)
- A Simpler Grid CSS
- Clean code

-----------------------------------------------------------------------------------------------------------
UPDATE instructions:
-----------------------------------------------------------------------------------------------------------

You can update the package via package installer, download the new version and place it in:
[base_path}/core/packages/
Update the package under "System" > "Package Managment"


-----------------------------------------------------------------------------------------------------------
IMPORTANT NOTICES FOR DEVELOPMENT:
-----------------------------------------------------------------------------------------------------------

- yepnope.js always executes things in the order they are listed. This is very important to know because it can hurt the speed of your website critically!
- If you need more than basic HTML5 support enabling with modernizr please pack your own: http://modernizr.github.com/Modernizr/2.0-beta/ I only applied Audio,Cideo,Canvas,appCache,postMessage,InputTypes,InputAttributes.  No GeolocAPI,SVG,WebGL, SMIL support is enabled, too.
- for how-to use the modernizr please read this: http://blogs.sitepoint.com/2011/03/08/regressive-enhancement-with-modernizr-and-yepnope/
+ added monderizr blog entry
- If you need more than basic HTML5 support enabling with modernizer please pack your own: http://modernizr.github.com/Modernizr/2.0-beta/ I only applied Audio,Cideo,Canvas,appCache,postMessage,InputTypes,InputAttributes.  No GeolocAPI,SVG,WebGL, SMIL support is enabled, too.
- If you want to add CSS3 behavior to IE<9 you can use the PIE.htc. You just have to add behavior: url(path/to/PIE.htc); to your css properties. See here for more info: http://css3pie.com/documentation/getting-started/
- I provided support for different devices with @media-queries. Please consider to either use all that stylesheet links in "site_header.html" or remove them to avoid dead links!


-----------------------------------------------------------------------------------------------------------
CREDITS:
-----------------------------------------------------------------------------------------------------------

ALL supporters and contributors of the great HTML5Boilerplate:
- http://html5boilerplate.com/
- special thanks to: Paul Irish and Divya Manian for all they've done with HTML5Boilerplate
- https://github.com/paulirish/html5-boilerplate for the initial base and idea
- https://github.com/modernizr for MOdernizr
- the MODX Revolution team for the whole MODX system
- Anselm Hannemann for the MODx Boilerplate, http://anselm.novolo.de/


-----------------------------------------------------------------------------------------------------------
BUGS and FEATURE REQUESTS:
-----------------------------------------------------------------------------------------------------------

Please e-mail me at: info@designfromwithin.com

CHANGELOG.txt
Changelog for Flexibility

flexibility-2.0.6-alpha.transport (11-3-2012)
====================================
- Changed TV order
- Removed unwanted code
- Added Public License
- Remove includeTemplate code, now using Media Sources
- Updated packages (FormIt, getResources, simplesearch, TinyMCE)
- Updated Foundation Framework to 2.2
- Fixed small dropdown issue

flexibility-2.0.5-alpha.transport (8-2-2012)
====================================
- Fixed sub-page-2 bug (thanks Showa!)

flexibility-2.0.4-alpha.transport (5-2-2012)
====================================
- Template variables are now added to the correct template.
- Ready for ALPHA testing!

flexibility-2.0.3-beta.transport (4-2-2012)
====================================
- Using just one category now.
- Fixed MIGX bug.

flexibility-2.0.2-beta.transport (17-12-2011)
====================================
- Added MIGX TV's for adding slides to the image slider.

flexibility-2.0.1-beta.transport (13-12-2011)
====================================
- First version online!

_build/data/transport.tvs.php
<?php
/**
* Template variable objects for the Flexibilitys package
* @author Menno Pietersen <info@designfromwithin.com>
* 13/12/2011
*
* @package flexibilitys
* @subpackage build
*/

/* Common 'type' options:
 * textfield (text line)
 * textbox
 * richtext
 * textarea
 * textareamini
 * email
 * html
 * image
 * date
 * option (radio buttons)
 * listbox
 * listbox-multiple
 * number
 * checkbox
 * tag
 * hidden
 */

/* Example template variables */

$templateVariables = array();

$templateVariables[1]= $modx->newObject('modTemplateVar');
$templateVariables[1]->fromArray(array(
    'id' => 1,
    'type' => 'richtext',
    'name' => 'footer_box_01',
    'caption' => 'Footer-content box 1',
    'description' => 'The content of the first footer-content box',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 15,
	'category' => 'Footer content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[2]= $modx->newObject('modTemplateVar');
$templateVariables[2]->fromArray(array(
    'id' => 2,
    'type' => 'richtext',
    'name' => 'footer_box_02',
    'caption' => 'Footer-content box 2',
    'description' => 'The content of the second footer-content box',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 16,
	'category' => 'Footer content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[3]= $modx->newObject('modTemplateVar');
$templateVariables[3]->fromArray(array(
    'id' => 3,
    'type' => 'richtext',
    'name' => 'footer_box_03',
    'caption' => 'Footer-content box 3',
    'description' => 'The content of the third footer-content box',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 17,
	'category' => 'Footer content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[4]= $modx->newObject('modTemplateVar');
$templateVariables[4]->fromArray(array(
    'id' => 4,
    'type' => 'richtext',
    'name' => 'footer_box_04',
    'caption' => 'Footer-content box 4',
    'description' => 'The content of the fourth footer-content box',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 17,
	'category' => 'Footer content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[5]= $modx->newObject('modTemplateVar');
$templateVariables[5]->fromArray(array(
    'id' => 5,
    'type' => 'richtext',
    'name' => 'page_box_01',
    'caption' => 'Sub-content box 1',
    'description' => 'The content of the first sub-content box',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 10,
	'category' => 'Page content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[6]= $modx->newObject('modTemplateVar');
$templateVariables[6]->fromArray(array(
    'id' => 6,
    'type' => 'richtext',
    'name' => 'page_box_02',
    'caption' => 'Sub-content box 2',
    'description' => 'The content of the second sub-content box',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 11,
	'category' => 'Page content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[7]= $modx->newObject('modTemplateVar');
$templateVariables[7]->fromArray(array(
    'id' => 7,
    'type' => 'richtext',
    'name' => 'page_box_03',
    'caption' => 'Sub-content box 3',
    'description' => 'The content of the third sub-content box',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 12,
	'category' => 'Page content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[8]= $modx->newObject('modTemplateVar');
$templateVariables[8]->fromArray(array(
    'id' => 8,
    'type' => 'richtext',
    'name' => 'page_box_04',
    'caption' => 'Sub-content box 4',
    'description' => 'The content of the fourth sub-content box',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 13,
	'category' => 'Page content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[9]= $modx->newObject('modTemplateVar');
$templateVariables[9]->fromArray(array(
    'id' => 9,
    'type' => 'richtext',
    'name' => 'page_lightbox',
    'caption' => 'Lightbox',
    'description' => 'The content of the optional lightbox for this page (leave empty to have no lightbox)',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 18,
	'category' => 'Page content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[10]= $modx->newObject('modTemplateVar');
$templateVariables[10]->fromArray(array(
    'id' => 10,
    'type' => 'richtext',
    'name' => 'sidebar',
    'caption' => 'Sidebar option',
    'description' => 'Content of the sidebar (leave empty if you want no sidebar)',
    'display' => 'default',
    'elements' => '',  /* input option values */
    'locked' => 0,
    'rank' => 1,
	'category' => 'Page content',
    'display_params' => '',
    'default_text' => '',
    'properties' => array(),
),'',true,true);

$templateVariables[11]= $modx->newObject('modTemplateVar');
$templateVariables[11]->fromArray(array(
    'id' => 11,
    'name' => 'optional_contactform',
    'caption' => 'Contactform',
    'description' => 'Do you want a contactform on this page?',
	'category' => 'Page options',
    'type' => 'option',
	'elements' => 'yes==yes||no==no',
    'display' => 'default',
    'locked' => 0,
    'rank' => 4,
    'default_text' => 'no',
    'display_params' => '',
	'properties' => array(),
),'',true,true);

$templateVariables[12]= $modx->newObject('modTemplateVar');
$templateVariables[12]->fromArray(array(
    'id' => 12,
    'name' => 'optional_gallery',
    'caption' => 'Gallery',
    'description' => 'Do you want a picture gallery on this page? (add pictures under: Components > Gallery)',
	'category' => 'Page options',
    'type' => 'option',
	'elements' => 'yes==yes||no==no',
    'display' => 'default',
    'locked' => 0,
    'rank' => 2,
    'default_text' => 'no',
    'display_params' => '',
	'properties' => array(),
),'',true,true);

$templateVariables[13]= $modx->newObject('modTemplateVar');
$templateVariables[13]->fromArray(array(
    'id' => 13,
    'name' => 'optional_slider',
    'caption' => 'Slider',
    'description' => 'Do you want a slider on this page? (too add slides go to: Site Settings > Template Variables)',
	'category' => 'Page options',
    'type' => 'option',
	'elements' => 'yes==yes||no==no',
    'display' => 'default',
    'locked' => 0,
    'rank' => 1,
    'default_text' => 'no',
    'display_params' => '',
	'properties' => array(),
),'',true,true);

$templateVariables[14]= $modx->newObject('modTemplateVar');
$templateVariables[14]->fromArray(array(
    'id' => 14,
    'name' => 'email_adress',
    'caption' => 'Email adress for the contactform',
    'description' => 'Select the email adress where the contactform should mail to',
	'category' => 'Site options',
    'type' => 'email',
	'elements' => '',
    'display' => 'default',
    'locked' => 0,
    'rank' => 3,
    'default_text' => '',
    'display_params' => '',
	'properties' => array(),
),'',true,true);

$templateVariables[15]= $modx->newObject('modTemplateVar');
$templateVariables[15]->fromArray(array(
    'id' => 15,
    'name' => 'FooterBoxNumber',
    'caption' => 'Footer boxes',
    'description' => 'Select the number of footer boxes',
	'category' => 'Site options',
    'type' => 'option',
	'elements' => '2==2||3==3||4==4',
    'display' => 'default',
    'locked' => 0,
    'rank' => 14,
    'default_text' => '2',
    'display_params' => '',
	'properties' => array(),
),'',true,true);

$templateVariables[16]= $modx->newObject('modTemplateVar');
$templateVariables[16]->fromArray(array(
    'id' => 16,
    'name' => 'siteLogo',
    'caption' => 'Logo',
    'description' => 'Select the website logo',
	'category' => 'Site options',
    'type' => 'image',
	'elements' => '',
    'display' => 'default',
    'locked' => 0,
    'rank' => 5,
    'default_text' => '',
    'display_params' => '',
	'properties' => array(),
),'',true,true);

$templateVariables[17]= $modx->newObject('modTemplateVar');
$templateVariables[17]->fromArray(array(
    'id' => 17,
    'name' => 'slidetext',
    'caption' => 'Richtext',
    'description' => 'Slide text',
    'category' => 'Enter the content of the slide',
    'type' => 'richtext',
    'display' => 'default',
    'locked' => 0,
    'rank' => 0,
    'default_text' => '',
    'display_params' => '',
),'',true,true);

$templateVariables[18]= $modx->newObject('modTemplateVar');
$templateVariables[18]->fromArray(array(
    'id' => 18,
    'name' => 'image',
    'caption' => 'image',
    'description' => 'Slider image',
	'category' => 'Site options',
    'type' => 'image',
    'display' => 'text',
    'locked' => 0,
    'rank' => 0,
    'default_text' => '',
    'display_params' => '',
),'',true,true);

$templateVariables[19]= $modx->newObject('modTemplateVar');
$templateVariables[19]->fromArray(array(
    'id' => 19,
    'name' => 'page_boxes',
    'caption' => 'Sub-content boxes',
    'description' => 'How many sub-content boxes do you want on this page?',
	'category' => 'Page options',
    'type' => 'option',
	'elements' => '0==0||2==2||3==3||4==4',
    'display' => 'default',
    'locked' => 0,
    'rank' => 8,
    'default_text' => '0',
    'display_params' => '',
	'properties' => array(),
),'',true,true);

$templateVariables[20]= $modx->newObject('modTemplateVar');
$templateVariables[20]->fromArray(array(
    'id' => 20,
    'name' => 'multiItemsGrid',
    'caption' => 'Slider items',
    'description' => 'The items in the slider',
	'category' => 'Site options',
    'type' => 'migx',
    'display' => 'default',
    'locked' => 0,
    'rank' => 7,
    'default_text' => '',
    'display_params' => '',
),'',true,true);

$input_properties=array(
    'formtabs' => '
[
{"caption":"Info", "fields": [
    {"field":"title","caption":"Title","description":"Title for the image."},
    {"field":"description","caption":"Description","inputTV":"slidetext"}
]},
{"caption":"Image", "fields":[
    {"field":"image","caption":"Image","inputTV":"image"}
]}
]
	',
    'columns' => '
[
{"header": "Title", "width": "160", "sortable": "true", "dataIndex": "title"},
{"header": "Image", "width": "50", "sortable": "false", "dataIndex": "image","renderer": "this.renderImage"}
]
');
$templateVariables[20]->set('input_properties',$input_properties);

return $templateVariables;